libs/api/domains/national-registry/src/lib/soap/nationalRegistryApi.ts

```diff
@@ -0,0 +1,386 @@
+    this.clientUser = clientUser
+    this.clientPassword = clientPassword
+  }
+
+  public async getMyInfo(nationalId: string): Promise<MyInfo | null> {
+    const response = await this.getViewThjodskra(nationalId)
+
+    if (!response)
+      throw new NotFoundException(
+        `user with nationalId ${nationalId} not found in national Registry`,
+      )
+
+    const userInfo = response.table.diffgram.DocumentElement.Thjodskra
+    const houseResponse = await this.getViewHusaskra(userInfo.LoghHusk)
+    const religionResponse = await this.getViewKennitalaOgTrufelag(nationalId)
+    const birthPlaceResponse = await this.getViewSveitarfelag(
+      userInfo.Faedsvfnr,
+    )
+    const banMarkingResponse = await this.getViewKennitalaOgBannmerking(
+      nationalId,
+    )
+
+    return {
+      fullName: userInfo.Nafn,
+      citizenship: userInfo.Rikisfang === 'IS' ? 'Ísland' : userInfo.Rikisfang,
+      gender: this.formatGender(userInfo.Kyn),
+      birthPlace: this.formatBirthPlaceString(birthPlaceResponse),
+      religion: this.formatReligionString(religionResponse),
+      legalResidence: this.formatResidenceAddressString(houseResponse),
+      maritalStatus: this.formatMartialStatus(userInfo.Hju, userInfo.Kyn),
+      banMarking: this.formatBanMarking(banMarkingResponse),
+    }
+  }
+
+  public async getMyFamily(nationalId: string): Promise<FamilyMember[] | null> {
+    const response = await this.getViewFjolskyldan(nationalId)
+    if (!response)
+      throw new NotFoundException(
+        `family for nationalId ${nationalId} not found`,
+      )
+
+    const family = response?.table.diffgram.DocumentElement.Fjolskyldan
+
+    if (!family)
+      throw new NotFoundException(
+        `family for nationalId ${nationalId} not found`,
+      )
+
+    const members = family.map(
+      (x) =>
+        ({
+          fullName: x.Nafn,
+          nationalId: x.Kennitala,
+          gender: x.Kyn,
+          maritalStatus: x.Hjuskapur,
+          address: `${x.Husheiti}, ${x.Pnr} ${x.Sveitarfelag}`,
+        } as FamilyMember),
+    )
+
+    return members
+  }
+
+  private formatResidenceAddressString(
+    houseDto: GetViewHusaskraDto | null,
+  ): string {
+    if (!houseDto) return 'address not found'
+    const houseInfo = houseDto.table.diffgram.DocumentElement.Husaskra
+    return `${houseInfo.HusHeiti}, ${houseInfo.PostNr} ${houseInfo.Nafn}`
+  }
+
+  private formatReligionString(
+    religtionDto: GetViewKennitalaOgTrufelag | null,
+  ): string {
+    if (!religtionDto) return 'religion info not found'
+
+    return religtionDto.table.diffgram.DocumentElement.KennitalaOgTrufelag
+      .Trufelag
+  }
+
+  private formatBirthPlaceString(
+    birthPlaceDto: GetViewSveitarfelagDto | null,
+  ): string {
+    if (!birthPlaceDto) return 'birth place not found'
+    return birthPlaceDto.table.diffgram.DocumentElement.Sveitarfelag.Sokn
+  }
+
+  private formatMartialStatus(maritalCode: string, genderCode: string): string {
```

Comment:
Might want to think about returning codes here so it can be localized in the frontend in the future

Reply:
Yeah definetly